internal/server/server.go

```diff
@@ -35,26 +38,47 @@ func requestHandler(ctx *fasthttp.RequestCtx) {
 	if host == "" {
 		ctx.Response.SetStatusCode(404)
 		ctx.Response.SetBodyString(fmt.Sprintf("Domain %s not found", ctx.Host()))
-	} else {
-		req.SetHost(host)
+	} else { // the route exists so we should have a deployment attached to the service
+		origin := store.GetRouteOrigin(host)
+		req.SetHost(origin)
 
+		// First try
 		if err := httpClient.Do(req, res); err != nil {
-			log.Error().Err(err).Msg("Error forwarding the request")
+			log.Debug().Msg("Error forwarding the request - Scaling up the deployment")
+			// Maybe the deployment is scaled down, let's scale it up
+			if err := kubernetes.ScaleUp(store.GetRouteLabel(host), store.GetRouteNamespace(host)); err != nil {
```

Comment:
Reading store twice? If planning to add locks/syncs here, better to get the whole object once then get what u need out of it. But this is def for optimization purposes. (Should have vs must have)